base_layer/core/src/proof_of_work/monero_rx/fixed_array.rs

```diff
@@ -109,33 +112,38 @@ impl ByteArray for FixedByteArray {
     }
 }
 
-impl Decodable for FixedByteArray {
-    fn consensus_decode<D: io::Read>(d: &mut D) -> Result<Self, encode::Error> {
-        #[allow(clippy::cast_possible_truncation)]
-        let len = VarInt::consensus_decode(d)?.0 as usize;
+impl ConsensusDecoding for FixedByteArray {
+    fn consensus_decode<R: Read>(reader: &mut R) -> Result<Self, io::Error> {
+        let mut buf = [0u8; 1];
+        reader.read_exact(&mut buf)?;
+        let len = buf[0] as usize;
         if len > MAX_ARR_SIZE {
-            return Err(encode::Error::ParseFailed(
-                "length exceeded maximum of 64-bytes for FixedByteArray",
+            return Err(io::Error::new(
+                io::ErrorKind::InvalidInput,
+                format!("length exceeded maximum of 64-bytes for FixedByteArray: {}", len),
             ));
         }
         let mut ret = FixedByteArray::new();
         for _ in 0..len {
             // PANIC: Cannot happen because len has been checked
-            ret.push(Decodable::consensus_decode(d)?);
+            let mut buf = [0u8; 1];
+            reader.read_exact(&mut buf)?;
+            ret.push(buf[0]);
         }
         Ok(ret)
     }
 }
 
-impl Encodable for FixedByteArray {
-    fn consensus_encode<E: io::Write>(&self, e: &mut E) -> Result<usize, io::Error> {
-        self.as_slice().consensus_encode(e)
+impl ConsensusEncoding for FixedByteArray {
+    fn consensus_encode<W: Write>(&self, writer: &mut W) -> Result<(), io::Error> {
+        writer.write_all(&[self.len][..])?;
+        writer.write_all(&self.elems[0..self.len()])?;
```

Comment:
Can make this encoding the same as the slice/vec<u8> encoding 
```suggestion
        self.as_slice().consensus_decode(writer)?;
```